### ISSUE TITLE
An area of interest representing a point fails the bbox query

### ISSUE THREAD

**Issue by snowman2** [OSGeo/proj.4#4236]
From: https://github.com/pyproj4/pyproj/issues/1326

#### Example of problem

Failing command with PROJ 9.2+
```
PROJ_DEBUG=3 cs2cs "WGS 84" "EPSG:32615" --bbox -93.581543,42.032974,-93.581543,42.032974
```
Failing output:
```
...
No operation found matching criteria
Rel. 9.4.1, June 1st, 2024
<cs2cs>: 
cannot initialize transformation
cause: (null)
program abnormally terminated
```
Successful command:
```
PROJ_DEBUG=3 cs2cs "WGS 84" "EPSG:32615" --bbox -93.581543,42.032974,-93.581542999,42.032974001
```

#### Problem description

An area of interest representing a point is failing the query.

#### Expected Output

Point queries work.

#### Environment Information

- PROJ version: 9.1.1 & 9.4.1 tested
- Operation System Information: Linux

#### Installation method

- conda

**Comment by jjimenezshaw** [comment]
Yes. I found it some time ago as well. I have the impression that is comes from the algorithm that "intersects" bounding boxes, that considers that tangent bounding boxes do not intersect. The consequence is that nothing intersects degenerated bounding box into point or line.

**Comment by snowman2** [comment]
Thanks @rouault 👍

**Comment by glostis** [comment]
In case this is of interest to someone landing on this issue like I did, I think that the commit that initially introduced the bug is https://github.com/OSGeo/PROJ/commit/07ef017f38471d66ff52b00fa802dfe7abf20f8c which was first included in PROJ 9.2.0.

The range of versions affected by this bug is therefore: `>= 9.2.0, < 9.5.0`
